File: src/main/java/com/catalin/vibelog/service/AuthService.java
package com.catalin.vibelog.service;

import com.catalin.vibelog.dto.request.LoginRequest;
import com.catalin.vibelog.dto.request.RegisterRequest;
import com.catalin.vibelog.dto.response.AuthResponse;

/**
 * Authentication operations: register new users and log in existing ones.
 */
public interface AuthService {

    /**
     * Register a new user account.
     *
     * @param dto the registration payload
     * @return an auth response containing JWT and user info
     * @throws com.catalin.vibelog.exception.EmailAlreadyExistsException    if email is taken
     * @throws com.catalin.vibelog.exception.UsernameAlreadyExistsException if username is taken
     */
    AuthResponse register(RegisterRequest dto);

    /**
     * Log in an existing user.
     *
     * @param dto the login payload
     * @return an auth response containing JWT and user info
     * @throws com.catalin.vibelog.exception.InvalidCredentialsException if email/password don’t match
     */
    AuthResponse login(LoginRequest dto);
}

File: frontend/src/api/authService.js
// src/api/authService.js
import api from './axios';

/**
 * Log in with { email, password }.
 * Returns { token, user } where:
 *  - token is the JWT
 *  - user  is the ProfileResponse/user object
 */
export const login = (credentials) =>
    api.post('/auth/login', credentials).then(res => res.data);

/**
 * Register with { email, username, password }.
 * Returns { token, user }.
 */
export const register = (credentials) =>
    api.post('/auth/register', credentials).then(res => res.data);
